go/integration_test/processor/create_test.go

```diff
@@ -633,7 +633,92 @@ func TestProcessor_CodeStartingWith0xEFCanNotBeCreated(t *testing.T) {
 					if result.GasUsed != sufficientGas {
 						t.Errorf("execution failed but gas was not fully used, used %d", result.GasUsed)
 					}
+				}
+			})
+		}
+	}
+}
+
+func TestProcessor_InitCodeSizeIsLimited(t *testing.T) {
+	maxInitCodeSize := int64(24576)
 
+	tests := map[string]struct {
+		revision tosca.Revision
+		length   *big.Int
+		success  bool
+	}{
+		"preShanghai": {
+			revision: tosca.R11_Paris,
+			length:   big.NewInt(maxInitCodeSize + 1),
+			success:  true,
+		},
+		"shanghai": {
+			revision: tosca.R12_Shanghai,
+			length:   big.NewInt(maxInitCodeSize + 1),
+			success:  false,
+		},
+		"shanghaiThreshold": {
+			revision: tosca.R12_Shanghai,
+			length:   big.NewInt(maxInitCodeSize),
+			success:  true,
+		},
+		"postShanghai": {
+			revision: tosca.R13_Cancun,
+			length:   big.NewInt(maxInitCodeSize + 1),
+			success:  false,
+		},
+	}
+
+	for processorName, processor := range getProcessors() {
+		for testName, test := range tests {
+			t.Run(processorName+"/"+testName, func(t *testing.T) {
+				sender := tosca.Address{1}
+				addressToBeCreated := tosca.Address(crypto.CreateAddress(common.Address(sender), 0))
+
+				initCode := []byte{byte(vm.PUSH2)}
+				initCode = append(initCode, test.length.Bytes()...)
```

Comment:
Could it be that this code is testing the size of the resulting contract instead of the size of the init code itself?

Reply:
Yes, I messed up the naming.